content/blog/2022/2022-02-14-spring-shell.md

```diff
@@ -0,0 +1,331 @@
+---
+title: "Create Command-line Applications with Spring Shell"
+categories: ["Spring"]
+date: 2022-02-26 00:00:00 +1100
+authors: ["cercenazi"]
+description: "Introduction to Spring Shell and how to create a simple command line application."
+image: images/stock/0119-cat-1920-1282.jpg
+url: spring-shell
+---
+
+## Spring Shell
+[Spring Shell](https://spring.io/projects/spring-shell) allows us to build a command line (shell) application using the Spring framework and all the advantages it provides.
+
+## What Is a Shell Anyways?
+A shell provides us with an interface to a system (usually an operating system) to which we give commands and parameters. The shell in turn does some useful tasks for us and provides an output.
+
+## Creating a Basic Shell
+First, we have to get the [SpringShell dependency](https://mvnrepository.com/artifact/org.springframework.shell/spring-shell-starter) from Maven central, which has everything we need.
+
+Then, since it's a Spring application, our main method has to be annotated with `@SpringBootApplication`
+```java  
+@SpringBootApplication 
+public class SpringShellApplication {    
+   public static void main(String[] args) {    
+      SpringApplication.run(SpringShellApplication.class, args);    
+	} 
```

Comment:
The code formatting of the code examples is inconsistent (i.e. some have the `{` in the same line, some have it in the next line. Please format the code consistently.